packages/crayons-extended/custom-objects/src/components/form-builder/components/field-editor.tsx

```diff
@@ -1096,14 +1403,235 @@ export class FieldEditor {
                 ? 'warning'
                 : 'normal'
             }
-            onFwBlur={this.internalNameBlurHandler}
-            onFwInput={this.internalNameInputHandler}
+            onFwBlur={(el) => this.internalNameBlurHandler(el, level)}
+            onFwInput={(el) => this.internalNameInputHandler(el, level)}
           ></fw-input>
         </div>
       </div>
     );
   }
 
+  private renderLabel(
+    objMaxLimits,
+    isDefaultNonCustomField,
+    boolEditAllowed,
+    fieldBuilderOptions
+  ) {
+    const objFieldBuilder = fieldBuilderOptions;
+    const strBaseClassName = 'fw-field-editor';
+    const strInputLabel = hasCustomProperty(objFieldBuilder, 'label')
+      ? objFieldBuilder.label
+      : '';
+    const boolDIsableInputLabel = isDefaultNonCustomField || !boolEditAllowed;
+
+    const strInputHint = this.isPrimaryField
+      ? i18nText('primaryFieldNameHint')
+      : '';
+    // Dependent Level checks
+    const level = fieldBuilderOptions?.field_options?.level;
+    const dictElName = this.isDependentField
+      ? `${this.DEP_LABEL_KEY}${level}`
+      : 'name';
+    const labelErrorMessage = this.isDependentField
+      ? this.dependentErrors[dictElName]
+      : this.labelErrorMessage;
+
+    const labelWarningMessage = this.isDependentField
+      ? this.dependentWarning[dictElName]
+      : this.labelWarningMessage;
+
+    const boolShowLabelError =
+      this.showErrors && labelErrorMessage && labelErrorMessage !== ''
+        ? true
+        : false;
+
+    const boolShowLabelWarning =
+      !boolShowLabelError && labelWarningMessage && labelWarningMessage !== ''
+        ? true
+        : false;
+
+    const strInputWarning = boolShowLabelWarning ? labelWarningMessage : '';
+    const strInputError = boolShowLabelError ? labelErrorMessage : '';
+    const numLabelMaxChars = objMaxLimits?.['maxLabelChars']?.count || 255;
+    const fieldLabel = this.isDependentField
+      ? i18nText('labelForLevel', { level })
+      : i18nText('fieldLabel');
+
+    return (
+      <fw-input
+        ref={(el) => (this.dictInteractiveElements[dictElName] = el)}
+        class={`${strBaseClassName}-content-required-input`}
+        placeholder={i18nText('fieldLabelPlaceholder')}
+        label={fieldLabel}
+        required={true}
+        maxlength={numLabelMaxChars}
+        value={strInputLabel}
+        hintText={strInputHint}
+        errorText={strInputError}
+        warningText={strInputWarning}
+        state={
+          boolShowLabelError
+            ? 'error'
+            : boolShowLabelWarning
+            ? 'warning'
+            : 'normal'
+        }
+        disabled={boolDIsableInputLabel}
+        onFwBlur={(el) => this.labelBlurHandler(el, level)}
+        onFwInput={(el) => this.labelInputHandler(el, level)}
+      ></fw-input>
+    );
+  }
+
+  private renderFieldContent(
```

Comment:
Can we move these new render methods into a new component with some name like fb-dependent-editor.tsx or something as this file size will become huge and difficult to maintain

Reply:
All the dependent field related renderring methods can be moved to a separate file

Reply:
I will take this as a separate PR post this review.